manifests/init.pp

```diff
@@ -76,6 +80,25 @@
   )
   validate_integer($choco_install_timeout_seconds)
   validate_bool($enable_autouninstaller)
+  validate_string($proxy_server)
+  validate_string($proxy_username)
+  validate_string($proxy_password)
+
+  if $proxy_port {
+    validate_integer($proxy_port)
+  }
+  
+  if $proxy_server or $proxy_port {
+    if ( ! $proxy_server) or ( ! $proxy_port) {
+      fail('Both proxy_port and proxy_server must be defined!')
+    }
+  }
+
+  if $proxy_username or $proxy_password {
+    if ( ! $proxy_password) or ( ! $proxy_username) or ( ! $proxy_server) {
+      fail('If proxy username or password is specified, all proxy settings must be defined!')
+    }
+  }
```

Comment:
That's pretty nice. The specs will help drive out whether this is correct or not. :+1: